src/components/KFileUpload/KFileUpload.vue

```diff
@@ -0,0 +1,344 @@
+<template>
+  <div
+    class="k-file-upload w-100"
+  >
+    <KLabel
+      v-if="label && appearance === 'file'"
+      :for="customInputId"
+      v-bind="labelAttributes"
+      data-testid="k-file-upload-label"
+    >
+      {{ label }}
+    </KLabel>
+
+    <KInput
+      :id="customInputId"
+      ref="fileInput"
+      :key="fileInputKey"
+      type="file"
+      :accept="accept"
+      :help="help"
+      :max-file-size="maximumFileSize"
+      :class="[
+        'w-100',
+        'upload-input',
+        'cursor-pointer',
+        { 'image-upload': appearance === 'image' },
+        { 'input-error' : hasUploadError }
+      ]"
+      @change="onFileChange"
+    />
+
+    <slot name="icon">
```

Comment:
I'm not sure we need to give them the option of slotting this. We offer so many icons now they should be able to find what they want. What do you think?